int32 ik_data
